examples/blame_example.ipynb

```diff
@@ -68,211 +62,270 @@
+    "scene_plotter = iai.utils.ScenePlotter(rendered_static_map,\n",
+    "                                       location_info_response.map_fov,\n",
+    "                                       (location_info_response.map_center.x, location_info_response.map_center.y),\n",
     "                                       location_info_response.static_actors)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
-   "metadata": {},
+   "execution_count": 6,
+   "metadata": {
+    "id": "L0ufxMO8NdvX",
+    "scrolled": true
+   },
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "[((0, 3), (89, 96), (0,)), ((2, 17), (135, 150), (2,)), ((3, 5), (150, 162), (3,)), ((4, 11), (1, 15), (11,)), ((4, 21), (61, 61), (21,)), ((11, 21), (16, 29), (21,)), ((12, 17), (193, 196), (12,)), ((17, 22), (120, 121), (22,))]\n"
+      "[LogCollision(collision_agents=(2, 4), start_time=85, end_time=88)]\n"
      ]
     }
    ],
    "source": [
-    "collisions = None\n",
+    "@dataclass\n",
+    "class LogCollision:\n",
+    "    collision_agents: Tuple[int, int]\n",
+    "    start_time: int \n",
+    "    end_time: int\n",
+    "\n",
+    "def compute_pairwise_collisions(agent_states_history,agent_attributes):\n",
+    "    \"\"\"\n",
+    "    Use polygon intersections to check each agent combination whether there is a collision.\n",
+    "    \n",
+    "    Args:\n",
+    "        agent_states: List of all current agent states including x and y coordinates and angle.\n",
+    "        agent_attributes: List of static attributes of all agents including agent length and width.\n",
+    "\n",
+    "    Returns:\n",
+    "        List of colliding pairs of agents. => List[List[int,int]]\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    collisions_ongoing = {}\n",
+    "    collisions_all = []\n",
+    "    \n",
+    "    #Iterate through the entire agent state history to find collisions at each time step\n",
+    "    for t, agent_states in enumerate(agent_states_history):\n",
+    "        num_agents = len(agent_states)\n",
+    "        if not num_agents == len(agent_attributes):\n",
+    "            raise Exception(\"Incorrect number of agents or agent attributes.\")\n",
+    "\n",
+    "        #Transform the points of the agents polygon into the common world coordinate frame\n",
+    "        polygons = [None]*num_agents\n",
+    "        for i, (state, attributes) in enumerate(zip(agent_states,agent_attributes)):\n",
+    "            dx = attributes.length/2\n",
+    "            dy = attributes.width/2\n",
+    "\n",
+    "            origin = np.array([[state.center.x], [state.center.y]])\n",
+    "\n",
+    "            angle = state.orientation\n",
+    "            c, s = np.cos(angle), np.sin(angle)\n",
+    "            \n",
+    "            #Rotation matrix\n",
+    "            R = np.array([[c, -s],\n",
+    "                          [c, s]])\n",
```

Comment:
I'm not totally sure what this question means but if you're asking why I used a numpy array instead of a list, even though matmul() can operate without numpy arrays, this is done for cleanliness and future-proofing.

Reply:
Oh I see, I put the cosine in the wrong element. This has been corrected.